plugins/woocommerce/src/StoreApi/Formatters/MoneyFormatter.php

```diff
@@ -23,12 +24,24 @@ public function format( $value, array $options = [] ) {
 			]
 		);
 
-		return (string) intval(
-			round(
-				( (float) wc_format_decimal( $value ) ) * ( 10 ** absint( $options['decimals'] ) ),
-				0,
-				absint( $options['rounding_mode'] )
-			)
-		);
+		// Ensure rounding mode is valid.
+		$rounding_modes           = [ PHP_ROUND_HALF_UP, PHP_ROUND_HALF_DOWN ];
+		$options['rounding_mode'] = absint( $options['rounding_mode'] );
+		if ( ! in_array( $options['rounding_mode'], $rounding_modes, true ) ) {
+			$options['rounding_mode'] = PHP_ROUND_HALF_UP;
+		}
+
+		// This ensures we work with floats from this point on. Also, if $value skips this as an array/object,
+		// it will throw a fatal error or produce an E_NOTICE (if Object on PHP <8.0).
```

Comment:
My rationale for this was avoiding converting arrays/objects to floats since that can result in the 1/0 value prices. But since the original param type was soft declared as mixed, and this is a public API you are right about this.

Inspecting the original code:

`( (float) wc_format_decimal( $value ) ) * ( 10 ** absint( $options['decimals'] ) ),`

This is where $value is used, consumed by `wc_format_decimal()`. First use is with:

```php
if ( ! is_float( $number ) ) {
    $number = str_replace( $decimals, '.', $number );

    // Convert multiple dots to just one.
    $number = preg_replace( '/\.(?![^.]+$)|[^0-9.-]/', '', wc_clean( $number ) );
}
```

And `str_replace`, `preg_replace` and `wc_clean` accept arrays. No fatal so far.

All the next if causes would be avoided and it would return an array, which would be converted to `1` after the (float) typecast.

It would fatal with another param type passed in. 

So to sum up I'm inclined to add `is_array()` to the if and recheck params for `wc_format_decimal`, to keep the same behaviour regarding fatals/wrong value prices due to array > float conversions, and add a doing_it_wrong if the param is not the one newly listed. And remove mixed from the method sig.